src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGenYul.scala

```
@@ -0,0 +1,129 @@
+            case None =>
+                Paths.get(mainName)
+        }
+
+        // temporary code, not designed to be recursive
+        val mainContract = mainContractOption.get
+        mainContract match {
+            case obsContract: ObsidianContractImpl => {
+                val translated_obj =  translateContract(obsContract)
+                yulString(translated_obj, finalOutputPath)
+            }
+            case _ => None
+        }
+//        val ast_seq =  translateProgram(ast)
+        return true
+    }
+
+    def translateProgram(program: Program): Unit = {
+        // ignore imports for now
+        for (c <- program.contracts) {
+            c match {
+                case obsContract: ObsidianContractImpl => translateContract(obsContract)
+                case javaContract: JavaFFIContractImpl => None
+            }
+        }
+    }
+
+    def translateContract(contract: ObsidianContractImpl): Object = {
+        // create runtime obj
+        var subobjects: Seq[Object] = Seq()
+        val runtime_name = contract.name + "_deployed"
+        val runtime_obj = Object(runtime_name, Code(Block(Seq())), Seq(), Seq())
+        subobjects = runtime_obj +: subobjects
+
+        var statement_seq: Seq[YulStatement] = Seq()
+        // initialization
+//        statement_seq = statement_seq +: translate_mem_init() // TODO check seq append
+
+        // declaration
+        // what to do when instance of constructor
+        for (decl <- contract.declarations if !decl.isInstanceOf[Constructor]) {
+//            statement_seq = statement_seq +: translateDeclaration(decl)
+        }
+
+        val create = Object(contract.name, Code(Block(statement_seq)), subobjects, Seq())
+        return create
+    }
+
+//    def translate_mem_init(): Seq[Statement]
+
+    def translateDeclaration(declaration: Declaration): Seq[YulStatement] = {
+        declaration match {
+            case f: Field => translateField(f)
+            case t: Transaction =>Seq()
```

Give warnings that these are unimplemented instead of failing silently.